src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php

```diff
@@ -73,7 +72,21 @@ public function dispatchEvent($eventName, EventArgs $eventArgs = null)
      */
     public function getListeners($event = null)
     {
-        return $event ? $this->listeners[$event] : $this->listeners;
+        if (null !== $event) {
+            if (!isset($this->initialized[$event])) {
+                $this->initializeListeners($event);
+            }
+
+            return $this->listeners[$event];
+        }
+
+        foreach (array_keys($this->listeners) as $event) {
```

Comment:
calling array_keys is unneeded extra CPU:
`foreach ($this->listeners as $event => $listeners) {`